packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5074,18 +5085,30 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   }
 
   @override
-  InheritedWidget dependOnInheritedElement(InheritedElement ancestor, {Object? aspect}) {
+  InheritedWidget dependOnInheritedElement(
+    InheritedElement ancestor, {
+    Object? aspect,
+    bool trackForCleanup = false,
```

Comment:
This API may introduce some bug that let's say there are two dependencies [A, B]
where B is called with trackForCleanup to false, and A is called with trackForCleanup to true.

B will end up get removed.

It feels like this API relies on all of the dependencies are called with trackForCleanup = true or all called with trackForCleanup = false.

Should we rewrite the mechanism to be per inherited element setting?